### ISSUE TITLE
[incubator/elasticsearch] Upgrade default elastic version

### ISSUE THREAD

**Issue by calbach** [helm/charts#1864]
FEATURE_REQUEST: Upgrade elasticsearch chart's default Elastic Search version from 2.4 -> 5.5.

**Which chart**:
incubator/elasticsearch

The chart defaults to a docker image with an old Elastic Search version (2.4), but there may be topological changes which would need to be applied along with the version change (I'm not sure).

**Comment by calbach** [comment]
FWIW I tried naively upgrading the docker image to use Elastic 5.5.1 with the [fabric8 plugin](https://github.com/fabric8io/elasticsearch-cloud-kubernetes) of the same version, and my 2 master nodes crash-looped on their readiness checks failing to find eachother. I didn't spend too much effort debugging this, but including my notes here in case someone else finds it useful.

Dockerfile:
```
FROM docker.elastic.co/elasticsearch/elasticsearch:5.5.1

RUN elasticsearch-plugin install io.fabric8:elasticsearch-cloud-kubernetes:5.5.1
```

`bootstrap.mlockall` was no longer recognized, so I had to comment that out:
```
--- a/incubator/elasticsearch/templates/elasticsearch-configmap.yaml
+++ b/incubator/elasticsearch/templates/elasticsearch-configmap.yaml
@@ -16,7 +16,7 @@ data:
     node.name: ${HOSTNAME}
 
     # see https://github.com/kubernetes/kubernetes/issues/3595
-    bootstrap.mlockall: ${BOOTSTRAP_MLOCKALL:false}
+    # bootstrap.mlockall: ${BOOTSTRAP_MLOCKALL:false}
```

**Comment by davidvonthenen** [comment]
Looks like there is a pending PR https://github.com/kubernetes/charts/pull/1794/ updating ES to 5.4.

**Comment by icereval** [comment]
FYI, the PR has been merged, I've also made an ES `5.5` build [available](https://hub.docker.com/r/centerforopenscience/elasticsearch/tags/) on docker hub.
